editor/components/block-drop-zone/index.js

```diff
@@ -53,10 +54,35 @@ function BlockDropZone( { index, isLocked, ...props } ) {
 		}
 	};
 
+	const reorderBlock = ( event, position ) => {
+		if ( index !== undefined && event.dataTransfer ) {
+			try {
+				const { uid, fromIndex, type } = JSON.parse( event.dataTransfer.getData( 'text' ) );
+
+				if ( type !== BLOCK_REORDER ) {
+					return;
+				}
+
+				if ( position.y === 'top' && index > fromIndex ) {
+					props.onDrop( event, uid, index - 1 );
+					return;
+				} else if ( position.y === 'bottom' && index < fromIndex ) {
+					props.onDrop( event, uid, index + 1 );
+					return;
+				}
+
+				props.onDrop( event, uid, index );
+			} catch ( err ) {
+				// console.log( err );
```

Comment:
Reminder to ask for feedback on improving the error handling here.

Reply:
In which case this is triggered? I believe when calling `JSON.parse`? I'm not familiar with Drag and Drop API so wondering if the dataTransfer could hold objects instead of strings to avoid having to parse it?

If not, may be we can only wrap the parsing by try/catch.

Reply:
> I'm not familiar with Drag and Drop API so wondering if the dataTransfer could hold objects instead of strings to avoid having to parse it?

The problem is IE only handles text in dataTransfer :/

> If not, maybe we can only wrap the parsing by try/catch.

Yeap, will refactor with that in mind.